public/app/features/alerting/unified/rule-editor/formDefaults.ts

```diff
@@ -0,0 +1,157 @@
+
+  return Object.freeze({
+    name: '',
+    uid: '',
+    labels: [{ key: '', value: '' }],
+    annotations: defaultAnnotations,
+    dataSourceName: GRAFANA_RULES_SOURCE_NAME, // let's use Grafana-managed alert rule by default
+    type: canCreateGrafanaRules ? RuleFormType.grafana : canCreateCloudRules ? RuleFormType.cloudAlerting : undefined, // viewers can't create prom alerts
+    group: '',
+
+    // grafana
+    folder: undefined,
+    queries: [],
+    recordingRulesQueries: [],
+    condition: '',
+    noDataState: GrafanaAlertStateDecision.NoData,
+    execErrState: GrafanaAlertStateDecision.Error,
+    evaluateFor: DEFAULT_GROUP_EVALUATION_INTERVAL,
+    evaluateEvery: DEFAULT_GROUP_EVALUATION_INTERVAL,
+    manualRouting: getDefautManualRouting(), // we default to true if the feature toggle is enabled and the user hasn't set local storage to false
+    contactPoints: {},
+    overrideGrouping: false,
+    overrideTimings: false,
+    muteTimeIntervals: [],
+    editorSettings: getDefaultEditorSettings(),
+
+    // cortex / loki
+    namespace: '',
+    expression: '',
+    forTime: 1,
+    forTimeUnit: 'm',
+  });
+};
+
+export const getDefautManualRouting = () => {
+  // first check if feature toggle for simplified routing is enabled
+  const simplifiedRoutingToggleEnabled = config.featureToggles.alertingSimplifiedRouting ?? false;
+  if (!simplifiedRoutingToggleEnabled) {
+    return false;
+  }
+  //then, check in local storage if the user has enabled simplified routing
+  // if it's not set, we'll default to true
+  const manualRouting = localStorage.getItem(MANUAL_ROUTING_KEY);
+  return manualRouting !== 'false';
+};
+
+function getDefaultEditorSettings() {
+  const editorSettingsEnabled = config.featureToggles.alertingQueryAndExpressionsStepMode ?? false;
+  if (!editorSettingsEnabled) {
+    return undefined;
+  }
+  //then, check in local storage if the user has saved last rule with sections simplified
+  const queryEditorSettings = localStorage.getItem(SIMPLIFIED_QUERY_EDITOR_KEY);
+  const notificationStepSettings = localStorage.getItem(MANUAL_ROUTING_KEY);
+  return {
+    simplifiedQueryEditor: queryEditorSettings !== 'false',
+    simplifiedNotificationEditor: notificationStepSettings !== 'false',
+  };
+}
+
+export function formValuesFromQueryParams(ruleDefinition: string, type: RuleFormType): RuleFormValues {
+  let ruleFromQueryParams: Partial<RuleFormValues>;
+
+  try {
+    ruleFromQueryParams = JSON.parse(ruleDefinition);
+  } catch (err) {
+    return {
+      ...getDefaultFormValues(),
+      queries: getDefaultQueries(),
+    };
+  }
+
+  return setQueryEditorSettings(
+    setInstantOrRange(
+      revealHiddenQueries({
+        ...getDefaultFormValues(),
+        ...ruleFromQueryParams,
+        annotations: normalizeDefaultAnnotations(ruleFromQueryParams.annotations ?? []),
+        queries: ruleFromQueryParams.queries ?? getDefaultQueries(),
```

Comment:
I think all invocations of `getDefaultQueries` should take the parameter (like in line 139) but the `type` parameter can't be trusted in its current form